### ISSUE TITLE
Authenticated requests with a config object throwing error

### ISSUE THREAD

**Issue by jeff-kilbride** [Sainglend/tda-api-client#14]
Thanks for the module!

I'm getting a "no such file or directory" error when I supply an `authConfig` key in the endpoint configuration object. I just upgraded to 2.x and this wasn't happening in 1.1.2. Here is a simple script I am using to test the `getTransactions` endpoint:

```javascript
import path from 'path';
import arg from 'arg';
import { getTransactions, ETransactionType } from 'tda-api-client';
import {
  format as formatDate,
  parseISO as parseISODate,
  isBefore as isBeforeDate,
} from 'date-fns';

const args = arg({
  // Types
  '--type': String,
  '--symbol': String,
  '--startDate': String,
  '--endDate': String,

  // Aliases
  '-t': '--type',
  '-s': '--symbol',
  '--start': '--startDate',
  '--end': '--endDate',
});

const config = {
  accountId: '123456789',
  // authConfigFileLocation: path.resolve('../config/tdaClientAuth.json'),
  authConfig: {
    refresh_token: 'XXXXXXXXX',
    client_id: 'EXAMPLE@AMER.OAUTHAP',
  },
};

if (args['--type']) {
  if (ETransactionType[args['--type']]) {
    config.type = args['--type'];
  } else {
    console.error(`\n\tInvalid transaction type: ${args['--type']}`);
    process.exit(-1);
  }
}

if (args['--symbol']) {
  config.symbol = args['--symbol'];
}

const reDate = /\d{4}-\d{2}-\d{2}/;
if (args['--startDate']) {
  if (reDate.test(args['--startDate'])) {
    config.startDate = args['--startDate'];
  } else {
    console.error(
      `\n\tstartDate must be in YYYY-MM-DD format: ${args['--startDate']}`
    );
    process.exit(-1);
  }
}

if (args['--endDate']) {
  if (reDate.test(args['--endDate'])) {
    config.endDate = args['--endDate'];
  } else {
    console.error(
      `\n\tendDate must be in YYYY-MM-DD format: ${args['--endDate']}`
    );
    process.exit(-1);
  }
}

if (config.startDate && config.endDate) {
  const sDate = parseISODate(config.startDate);
  const eDate = parseISODate(config.endDate);
  if (!isBeforeDate(sDate, eDate)) {
    console.error(
      `\n\tstartDate (${config.startDate}) ` +
        `must come before endDate (${config.endDate})`
    );
    process.exit(-1);
  }
}

const result = await getTransactions(config);

console.log(result);
```
Obviously, I've removed the valid values for my account info. Running the script with the `authConfig` key throws an error:

```
$ node getTransactions.js -s IBM
node:internal/process/esm_loader:94
    internalBinding('errors').triggerUncaughtException(
                              ^

[Error: ENOENT: no such file or directory, open '/Users/jeff/VSCode/TDA/config/tdaclientauth.json'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/jeff/VSCode/TDA/config/tdaclientauth.json'
}
```
It seems the module is looking for the `tdaclientauth.json` file in the default location, even though I'm providing the `authConfig` key. If I uncomment the `authConfigFileLocation` key and comment out the `authConfig` key, everything works as expected.

I've looked through the `getAPIAuthentication` method in the repo, as well as the transpiled code in my `node_modules` directory, but I don't see anything obvious. It looks like it should work. Let me know if I'm missing something!

**Comment by Sainglend** [comment]
Thanks for bringing this up. I see where the problem lies. The function that writes the output to the file isn't checking enough conditions. For now, here is a workaround (which includes another thing I need to mend):

Declare your config like this:
```javascript
const config = {
  accountId: '123456789',
  // authConfigFileLocation: path.resolve('../config/tdaClientAuth.json'),
  authConfig: {
    refresh_token: 'XXXXXXXXX',
    client_id: 'EXAMPLE@AMER.OAUTHAP',
  },
  authConfigFileAccess: "NONE",
};
```

If you are using typescript, you'll have to do a weird cast (sorry; I need to fix this too):
```typescript
const config = {
  accountId: '123456789',
  // authConfigFileLocation: path.resolve('../config/tdaClientAuth.json'),
  authConfig: {
    refresh_token: 'XXXXXXXXX',
    client_id: 'EXAMPLE@AMER.OAUTHAP',
  },
  authConfigFileAccess: "NONE" as ("NONE" | "READ_ONLY" | "READ_WRITE" | undefined),
};
```

By setting authConfigFileAccess to "NONE" you'll bypass the attempt to write to file.

**Comment by Sainglend** [comment]
This is now fixed in v2.1.1
